python/whylogs/core/datatypes.py

```diff
@@ -107,6 +107,57 @@ def _do_match(cls, dtype_or_type: Any, maybe_type: Optional[Any]) -> bool:
         return False
 
 
+class Tensor(DataType[Any]):
```

Comment:
The `Any` scares me :)

Reply:
This whole resolver scares me. Based on our standup, let's do some more design work before we get here.